### PR TITLE
Turn CCC off for Express and Prompt processing, for Run2 scenarios.

### PR DESCRIPTION
**Pull request by rovere**
The title says it all.
`scram b runtests` under Configuration/DataProcessing/test returns all successes.

@venturia @VinInn

### CODE REVIEW

**Comment by slava77** [comment]
Marco, please add a customization function some place in RecoTracker.Configuration to be able to apply the same in matrix workflows

**Comment by rovere** [comment]
Ciao Slava, I already replied to you via email and I'm wondering if those 2 lines could be "manually" passed to cmsDriver to produce the RelVal. A customise for this seems to be an overkill.
If there is no alternative, I can easily prepare it.

**Comment by slava77** [comment]
I prefer a function.

File: RecoTracker/Configuration/python/customiseNoCCC.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import FWCore.ParameterSet.Config as cms
+
+def customiseNoCCC(process):
+
+    # apply only in reco step
+    if not hasattr(process,'reconstruction'):
```
Comment (slava77) [inline RecoTracker/Configuration/python/customiseNoCCC.py]: nice, thanks

**Comment by slava77** [comment]
@cmsbuild please test
I think that the customiseNoCCC can be propagated to 75X/76X as well

**Comment by VinInn** [comment]
unrelated:
maybe we ca get rid of "planBTracking"
it does not apply anymore as is...

**Comment by slava77** [comment]
On 7/1/15 11:40 AM, Vincenzo Innocente wrote:
> unrelated:
> maybe we ca get rid of "planBTracking"
> it does not apply anymore as is...
> 

I don't even know what it is ;)
Is it a tracking version that was added before the big bang?

**Comment by VinInn** [comment]
@slava77 
most probably in a parallel (B) universe

**Comment by slava77** [comment]
what's happening in HLT? Should the CCC be disabled there as well in the 50ns?
@Martin-Grunewald

**Comment by Martin-Grunewald** [comment]
@slava77 
Good question!- Can you explain why you switch this off?

**Comment by slava77** [comment]
@Martin-Grunewald 
the intention is to disable it for 50ns running in prompt/express reco.
This is triggered by concerns in the tracker calibrations quality.

**Comment by slava77** [comment]
it looks like HLT is not using these in the 50ns tables. So, no action is needed for HLT

**Comment by Martin-Grunewald** [comment]
@slava77 
OK, good to know. Thanks!

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/9987 3fd3676
- the customization function works at the runTheMatrix --command level

I ran comparisons with this more disabled: there are differences in the usual suspects, as detailed below. I'd say that for most cases we can live with a similar level data-MC discrepancy.

- strip-based seeding
    - in 50202 (PU35 at 50 ns MC) about 20% more seeds in mixed triplet and pixel-less and about 40% more in the tob-tec iterations
- tracking fake rates
    - high purity fakes in the barrel up by ~50%
    - general tracks fake rate increase is smaller (more visible increase at higher PU)
- dedx distributions
Nothing particularly significant beyond this.
The following plots are from 50202 DQM (400 events run)
wf50202hs_csvv2_roc
wf50202hs_general_algo
wf50202hs_general_dedx1
wf50202hs_general_dups_eta
wf50202hs_general_eff_eta
wf50202hs_general_eff_vtxpos
wf50202hs_general_fake_pu
wf50202hs_general_fr_vs_eta
wf50202hs_hp_fr_vs_eta

wf50202hs_general_tobhits_eta


On the technical side (based on 50202)
- reco step time goes up by 7%
```
  delta/mean delta/orJob     original                   new       module name
  ---------- ------------     --------                  ----       ------------
   +0.609220      +0.34%       228.74 ms/ev ->       429.13 ms/ev tobTecStepTrackCandidates
   +0.456152      +0.30%       299.11 ms/ev ->       475.87 ms/ev pixelLessStepTrackCandidates
   +0.399381      +0.06%        66.55 ms/ev ->        99.76 ms/ev tobTecStepSeedsTripl
   +0.242304      +0.05%       110.02 ms/ev ->       140.35 ms/ev pixelLessStepTracks
   +0.182235      +0.15%       440.42 ms/ev ->       528.72 ms/ev pixelLessStepSeeds
   +0.164758      +0.11%       349.56 ms/ev ->       412.33 ms/ev lowPtTripletStepTrackCandidates
   +0.121433      +0.12%       558.80 ms/ev ->       631.05 ms/ev detachedTripletStepTrackCandidates
```

**Comment by slava77** [comment]
@davidlange6 are we waiting for some additional re-confirmation that this is good to go for prompt/express?

**Pull request merged**